compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1227,7 +1227,18 @@ static LogicalResult setDefaultGenericOpRootConfig(
                                  parallelTileSizes, reductionTileSizes);
 
   TileSizesListType tileSizes;
-  tileSizes.push_back(flowTileSizes);
+  SmallVector<int64_t> updatedFlowTileSizes;
+  bool foundNonZero = false;
+  for (auto val : flowTileSizes) {
+    if (val == 0) {
+      updatedFlowTileSizes.push_back(foundNonZero || tileSizes.size() <= 3 ? 0
+                                                                           : 1);
+      continue;
+    }
+    updatedFlowTileSizes.push_back(val);
+    foundNonZero = true;
+  }
+  tileSizes.push_back(updatedFlowTileSizes);
```

Comment:
I'm wondering if this is something we can do in `getDefaultDistributedLevelTileSizes` as the flow tile sizes are used by other utilities, such as `setX86WorkgroupTileSizes`.

Reply:
I am not sure what effects they have, so didn't them.

Reply:
That's kind of the big problem that we have with tile size selection right now: we start with some sizes that we are arbitrarily changing along the pipeline and different utilities end up seeing different value, which is not ok. If there isn't a super strong reason to do something ad-hoc, I would try hard to get this into `getDefaultDistributedLevelTileSizes`. IIUC, this should be applicable to any pipeline, right? If that's not the case, at the very least we should make sure that all the utilities consuming `flowTileSizes` see the same numbers.

Reply:
I can make the change and see if it works

Reply:
Moving it to `getDefaultDistributedLevelTileSizes` causes all the `materialize_*_launch_configuration.mlir` to fail

Reply:
Fail like crashing? If it's just a check mismatch, that should be fine as that test is testing the tile sizes that we generate... It should be just about replacing some `0`s with some `1`s?

Reply:
I fixed the tests.